zstash/hpss_utils.py

```diff
@@ -73,6 +75,7 @@ def add_file(tar, file_name):
         tarinfo.size = os.path.getsize(file_name)
     tar.addfile(tarinfo)
 
+    md5: Optional[str] = None
```

Comment:
>When you use a variable for the first time, mypy will essentially infer its type based on the very first assignment it sees.
One way of working around this limitation is to just forward-declare the variable with the expected type.
https://stackoverflow.com/a/52860650